### ISSUE TITLE
Support custom fields for the PushData struct

### ISSUE THREAD

**Issue by ramajd** [Nheko-Reborn/mtxclient#100]
as mentioned in the Matrix specification - [Client Server API](https://spec.matrix.org/v1.7/client-server-api/),  [`data`](https://github.com/Nheko-Reborn/mtxclient/blob/f4425af712afc6ad704a39b93c912432bd3c1914/include/mtx/requests.hpp#L436) parameter in `SetPusher` struct should be a dictionary of pusher specific parameters: 

> Required if kind is not null. A dictionary of information for the pusher implementation itself. If kind is http, this should contain url which is the URL to use to send notifications to.

working on push notification implementation for iOS devices, I found that there are a series of additional parameters we need to set in the [`PushData` struct](https://github.com/matrix-org/sygnal/blob/main/docs/applications.md#ios-applications-beware).  

```json
{
  "url": "https://push-gateway.location.here/_matrix/push/v1/notify",
  "format": "event_id_only",
  "default_payload": {
    "aps": {
      "mutable-content": 1,
      "content-available": 1,
      "alert": {"loc-key": "SINGLE_UNREAD", "loc-args": []}
    }
  }
}
```

so I'm proposing tho change the `data` field in `SetPusher` struct to support additional parameters (Maybe the `nlohmann::json`).

**Comment by deepbluev7** [comment]
No, that sounds reasonable. We generally try to avoid the nlohmann::json type in our API, as it is untyped and increases compile time a lot, but if we need to be that flexible, then there isn't really a way around it :)

**Comment by ramajd** [comment]
I also wanted to implement this custom payload as a `std::variant`. however, it only postpone the issue to a later time with different requirements.

so I will apply the modifications and submit a pull request then.
